Reconfigure qemu's build system as it is outside out a Xen tree

diff --git a/Rules.mk b/Rules.mk
new file mode 100644
index 0000000..afd2295
--- /dev/null
+++ b/Rules.mk
@@ -0,0 +1,27 @@
+INSTALL      = install
+INSTALL_DIR  = $(INSTALL) -d -m0755 -p
+INSTALL_DATA = $(INSTALL) -m0644 -p
+INSTALL_PROG = $(INSTALL) -m0755 -p
+
+CURSES_LIBS = -lncurses
+UTIL_LIBS = -lutil
+SONAME_LDFLAG = -soname
+SHLIB_CFLAGS = -shared
+
+CFLAGS += -D__XEN_TOOLS__
+
+# Enable implicit LFS support *and* explicit LFS names.
+CFLAGS  += $(shell getconf LFS_CFLAGS)
+CFLAGS  += -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
+LDFLAGS += $(shell getconf LFS_LDFLAGS)
+
+CFLAGS += -DNDEBUG
+CFLAGS += -O2 -fno-omit-frame-pointer
+
+# CFLAGS settings from xen.hg/tools/Rules.mk
+CFLAGS += -std=gnu99
+CFLAGS += -mno-tls-direct-seg-refs
+CFLAGS += -Wstrict-prototypes
+CFLAGS += -Wno-unused-value
+CFLAGS += -m64
+CFLAGS += -Wdeclaration-after-statement
diff --git a/xen-config-host.mak b/xen-config-host.mak
index aea181f..b266f37 100644
--- a/xen-config-host.mak
+++ b/xen-config-host.mak
@@ -1,6 +1,5 @@
 QEMU_ROOT ?= .
-XEN_ROOT ?= $(QEMU_ROOT)/../xen-unstable.hg
-include $(XEN_ROOT)/tools/Rules.mk
+include $(QEMU_ROOT)/Rules.mk
 
 ifdef CONFIG_STUBDOM
 export TARGET_DIRS=i386-stubdom
diff --git a/xen-config.mak b/xen-config.mak
index 77fef4f..71962c6 100644
--- a/xen-config.mak
+++ b/xen-config.mak
@@ -1,5 +1,4 @@
 QEMU_ROOT ?= ..
-XEN_ROOT ?= $(QEMU_ROOT)/../xen-unstable.hg
 
 include ../config-host.mak
 
